src/GitVersionCore/FileLocking/ILockFileApi.cs

```diff
@@ -0,0 +1,9 @@
+using System.IO;
+
+namespace GitVersion.FileLocking
+{
+    public interface ILockFileApi
```

Comment:
I don't quite understand the distinction between `IFileLocker` and`ILockFileApi`. Why are they separated? I'm also not a fan of using `Api` in the name of an interface. All C# interfaces (and public classes) are [application programming interfaces](https://en.wikipedia.org/wiki/Application_programming_interface). :)

Reply:
I decided to implement this in a composition flavor. When the "API" is about to have the ability to lock anything the FileLocker can lock a specific file that has the ability to lock one file but being callable by many concurrent calls to FileLocker.Lock(). So the file gets unlocked first when all FileLockUse instances are got disposed. So later you can graduate the use of the instance of FileLocker to your behave when you decide to go away from one global lock.